From 4378569f8f41d58313284b88aa6a54df3346546e Mon Sep 17 00:00:00 2001
From: "Fabio M. Di Nitto" <fdinitto@redhat.com>
Date: Wed, 25 Apr 2012 09:25:07 +0200
Subject: [PATCH] rgmanager: fix cpglockd vs rgmanager startup race
 condition with systemd

systemd consider a service fully operation as soon as /var/run/foo.pid is
written.

this creates a small window in which rgmanager can start before cpglockd
is operational and rgmanager would fail.

loop over cpglockdump to verify that cpglockd is functional and then
start rgmanager

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>
Reviewed-by: Lon Hohberger <lhh@redhat.com>
Reviewed-by: Ryan McCabe <rmccabe@redhat.com>
---
 rgmanager/init.d/rgmanager.in |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/rgmanager/init.d/rgmanager.in b/rgmanager/init.d/rgmanager.in
index fcf7698..e602ba0 100644
--- a/rgmanager/init.d/rgmanager.in
+++ b/rgmanager/init.d/rgmanager.in
@@ -79,6 +79,7 @@ start_cpglockd()
 {
 	rings="$(corosync-objctl 2>/dev/null |grep ringnumber | wc -l)"
 	[ "$rings" -gt "1" ] && service cpglockd start
+	while ! cpglockdump > /dev/null 2>&1; do sleep 1; done
 }
 
 rtrn=0
-- 
1.7.7.6

